src/builder.rs

```diff
@@ -295,51 +361,39 @@ impl Builder {
         note: Note,
         merkle_path: MerklePath,
     ) -> Result<(), SpendError> {
-        if !self.flags.spends_enabled() {
+        let (flags, anchor) = self.bundle_type.bundle_config();
+        if !flags.spends_enabled() {
             return Err(SpendError::SpendsDisabled);
         }
 
         // Consistency check: all anchors must be equal.
         let cm = note.commitment();
         let path_root = merkle_path.root(cm.into());
-        if path_root != self.anchor {
+        if path_root != anchor {
             return Err(SpendError::AnchorMismatch);
         }
```

Comment:
Idea: move this into a private helper method on `SpendInfo`. Construct `SpendInfo::new` first, and then call the helper passing in `anchor`. Once that passes, push into `self.spends`. This means the helper can then be reused in the new `bundle()` function, instead of being duplicated. (The helper method should include the zero-value gate that `bundle()` does).

Reply:
Yeah, I had considered that, this seems fine.